src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/RewindDataIndexStore.cs

```diff
@@ -44,10 +45,19 @@ public async Task InitializeAsync(IConsensus consensusParameters, ChainedHeader
 
             int heightToSyncTo = tip.Height > this.numberOfBlocksToKeep ? tip.Height - this.numberOfBlocksToKeep : 0;
 
-            for (int i = tip.Height; i >= heightToSyncTo; i--)
+            for (int i = tip.Height - 1; i >= heightToSyncTo; i--)
```

Comment:
Why the -1?

Reply:
We should not have rewind data of current tip, that was the reason there was a `continue;` there before, the continue is wrong because there should always be rewind data for past blocks, so better is to start form the right place and throw if not found.